OPAL/br/src/main/scala/org/opalj/br/Type.scala

```diff
@@ -1056,6 +1058,43 @@ final class ObjectType private ( // DO NOT MAKE THIS A CASE CLASS!
  */
 object ObjectType {
 
+    /**
+     *  Flushes the global cache for ObjectType instances. This does not include the predefined types,
+     *  which are kept in memory for performance reasons.
+     */
+    def flushTypeCache(): Unit = {
+        // First we need to write all cached new types to the actual array, otherwise
+        // we might delete the predefined types
+        updateObjectTypes()
+
+        val writeLock = cacheRWLock.writeLock()
+
+        writeLock.lock()
+        try {
+            // Find all keys in the Cache-Map that belong to ObjectTypes that are not predefined
+            val keysToRemove = mutable.HashSet[String]()
```

Comment:
Why not just mutable.Set?

Reply:
Actually, instead of collecting the keys, couldn't you just traverse `objectTypes` starting from HighestPredefinedTypeId and delete these?

Reply:
Yes that could also work... Just one doubt i have in mind, maybe you could comment on that:
There could be very rare thread interleavings where after `updateObjectTypes()` (which synchronizes `cache` and `objectTypes`) new OTs are created and added to `cache` before `flushTypeCache` acquires the lock. In those cases, this method you are proposing would not clear those new OTs from the cache (since they are not yet contained in `objectTypes`). This would create a problem, as the ID would be reset while there are still OTs with a higher ID in the `cache`, which would "materialize" into the `objectTypes` array on the next call to `updateObjectTypes()`. Do you think that's an actual problem?

Reply:
Agreed, that should be prevented. I think it should work if you move the writeLock before invoking updateObjectTypes?

Reply:
Yep that works, i just implemented it the way you suggested.